### PR TITLE
Add grayscull updates and pytest tests

### LINKED ISSUE #58: Pick up dependencies from pyproject.toml?
Proposal for a future enhancement:

SHAP now specifies its build and run dependencies in pyproject.toml, according to PEP 518. We could potentially pick up dependencies from there, rather than maintaining a separate list of build dependencies in this repo.

It looks like this would require `load_file_data` , which will be available in the next release of conda-build (probably out in a month or two).
https://github.com/conda/conda-build/pull/4480

### LINKED ISSUE #59: Run pytest on build?
It might be helpful to run the pytest suite against the built package, to ensure the c extension is built correctly.

The shap tests use pytest's `importorskip` to skip any tests requiring optional 3rd party dependencies (such as tensorflow or pytorch). These tend to be the long-running tests.

So, perhaps a good place to start would be just running the "core" pytest tests. I'd be happy to take a stab as part of #57 if you agree with the suggestion.

### PR DESCRIPTION
**Pull request by connortann**
Checklist
* [X] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [X] Bumped the build number (if the version is unchanged)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)

Fixes #58 , fixes #59

### CODE REVIEW

**Comment by connortann** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -42,8 +42,15 @@ requirements:
 test:
   requires:
     - pip
+    - pytest
+    - pytest-mpl
+    - matplotlib
```
Comment (xhochy) [inline recipe/meta.yaml:47]:
```suggestion
    - matplotlib-base
```

**Comment by connortann** [comment]
I was surprised to see such a variation in overall build times now that tests are included: most take ~7-10 minutes, but `ppc64` takes 1hr15 and `aarch64` takes 1hr50.

Maybe that's still acceptable though, as the build doesn't have to be run very often, and it's useful to test rigorously?

**Comment by xhochy** [comment]
`ppc64le` and `aarch64` run their tests through QEMU emulation. We can keep this as 1h is okish.

**Pull request merged**